src/__tests__/CompareResults/beta/RevisionSelect.test.tsx

```diff
@@ -1,12 +1,212 @@
+import ResultsView from '../../../components/CompareResults/beta/ResultsView';
 import RevisionSelect from '../../../components/CompareResults/beta/RevisionSelect';
+import useProtocolTheme from '../../../theme/protocolTheme';
 import { renderWithRouter } from '../../utils/setupTests';
-import { screen } from '../../utils/test-utils';
+import {
+  fireEvent,
+  renderHook,
+  screen,
+  waitFor,
+  within,
+} from '../../utils/test-utils';
+
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual<typeof import('react-router-dom')>('react-router-dom'),
+  useSearchParams: () => [new URLSearchParams({ fakedata: 'true' })],
+}));
 
 describe('Revision select', () => {
+  const protocolTheme = renderHook(() => useProtocolTheme()).result.current
+    .protocolTheme;
+  const toggleColorMode = renderHook(() => useProtocolTheme()).result.current
+    .toggleColorMode;
+
   it('Should match snapshot', () => {
     renderWithRouter(<RevisionSelect />);
 
     expect(screen.getByTestId('revision-select')).toBeInTheDocument();
     expect(document.body).toMatchSnapshot();
   });
+
+  it("Should change 'All revisions' option to bb6a5e451dac", async () => {
+    renderWithRouter(<RevisionSelect />);
+
+    let selectButton = await waitFor(() => screen.getByRole('button'));
```

Comment:
`await screen.findByRole` could be used here :-) 
Adnd the same is true for everywhere below, all the `waitFor` could be removed.

I'm not asking that you change all existing code, just your new code.